pkg/doc/verifiable/credential_test.go

```diff
@@ -157,6 +158,28 @@ func TestNew(t *testing.T) {
 		require.Error(t, err)
 		require.Contains(t, err.Error(), "JSON unmarshalling of verifiable credential failed")
 	})
+
```

Comment:
we should add an additional test case to provide an illustration of how a developer could choose which template to use.

1) decode into a basic verifiable credential
2) based on type info in the basic verifiable credential, decode into the more specialized type.

Reply:
I think there is the one at `extension_credential_test.go`. Do you this it's enough?

Reply:
@kdimak Does it show the above steps in sequence in a single test?

Reply:
@troyronda No, just because it's designed in a different way.
`udc := &UniversityDegreeCredential{}`
`cred, err := NewCredential(`
`	[]byte(validCredential),`
`	WithDecoders([]CredentialDecoder{udc.decode}),`
`	WithTemplate(udc.credential),`
`)`
We will get an error if the concrete VC is not supported by the VC extension.
I see your point here: in case we have several VC variants, we get basic VC and then decide what variant to use depending on the e.g. type (and context) of the basic VC.
I agree that it makes sense. Should we file a separate issue for this (as this goes beyond the current issue scope)?

Reply:
hey @troyronda 
I added this extra test case, see **extension_credential_switch_test.go**.
I did not add config of proof validations enablement as for now we are not supporting proofs at all.